repose-aggregator/services/service-client/impl/src/main/java/org/openrepose/core/services/serviceclient/akka/impl/AkkaServiceClientImpl.java

```diff
@@ -54,15 +55,16 @@
     private static final long FUTURE_CACHE_TTL = 500;
     private static final TimeUnit FUTURE_CACHE_UNIT = TimeUnit.MILLISECONDS;
     private static final int CONNECTION_TIMEOUT_BUFFER_MILLIS = 1000;
-    private final ServiceClient serviceClient;
     private final Cache<Object, Future> quickFutureCache;
+    private final HttpClientService httpClientService;
     private ActorSystem actorSystem;
-    private ActorRef tokenActorRef;
+    private volatile Timeout timeout;
+    private volatile ActorRef tokenActorRef;
```

Comment:
Because this doesn't post in chronological order. I want to avoid any `volatile` anything. If we end up there, we're doing concurrency wrong. We shouldn't need any `synchronized` either, we just need to find the right way to use the actor system in a way that makes sense. It might require more work than we thought, but if we don't and fall back to concurrency primitives we'll probably get it wrong. And it's a huge pain to get people to upgrade to newer versions of Repose, so we really need to do as much as possible to get it right the first time.